lib/iris/fileformats/_nc_load_rules/actions.py

```diff
@@ -385,13 +389,23 @@ def action_ukmo_stash(engine):
     attr_name = "ukmo__um_stash_source"
     attr_value = getattr(var, attr_name, None)
     if attr_value is None:
-        attr_altname = "um_stash_source"  # legacy form
-        attr_value = getattr(var, attr_altname, None)
+        attr_name = "um_stash_source"  # legacy form
+        attr_value = getattr(var, attr_name, None)
     if attr_value is None:
         rule_name += "(NOT-TRIGGERED)"
     else:
         # No helper routine : just do it
-        engine.cube.attributes["STASH"] = pp.STASH.from_msi(attr_value)
+        try:
+            stash_code = pp.STASH.from_msi(attr_value)
+        except (TypeError, ValueError):
+            engine.cube.attributes[attr_name] = attr_value
+            msg = (
+                "Unable to set attribute STASH as not a valid MSI "
+                f'string "mXXsXXiXXX", got "{attr_value}"'
+            )
+            logger.debug(msg)
+        else:
+            engine.cube.attributes["STASH"] = stash_code
```

Comment:
Similarly, I wasn't quite sure where this change originated.

Reply:
See the PR #4035, in particular [here](https://github.com/SciTools/iris/pull/4035/files#diff-5e278fdef6d1366f7c7c5b1256dca2ef8f9a30e5851a1c23b22a141fb787e183R1806).